### ISSUE TITLE
Doesnt work with "Tom Clancys Ghost Recon Future Soldier"

### ISSUE THREAD

**Issue by Illegal-Services** [baffler/Transparent-Twitch-Chat-Overlay#209]
It works fine but as soon as I start this game it doesn't work anymore.

Tested with TCO v0.9.5

**Comment by Illegal-Services** [comment]
Btw I tried the PowerToys trick from https://github.com/baffler/Transparent-Twitch-Chat-Overlay/issues/170
seems like it doesn't work as well

**Comment by Illegal-Services** [comment]
```py
import win32con
import win32gui

hwnd = 724210

while True:
    state = win32con.HWND_TOPMOST
    flags = win32con.SWP_NOMOVE | win32con.SWP_NOSIZE
    win32gui.ShowWindow(hwnd, win32con.SW_SHOW)
    win32gui.SetWindowPos(hwnd, state, 0,0,0,0, flags)
```

i tried this ugly solution in Python, it does work outside of the game, but when the game runs, it doesn't

**Comment by Illegal-Services** [comment]
![explorer_2023-10-20_12-36](https://github.com/baffler/Transparent-Twitch-Chat-Overlay/assets/62464560/a29a6fd4-9618-468c-9f7c-1c77289b4d77)

**Comment by Illegal-Services** [comment]
Actually about the previous post, I think it's pretty trash.
I've writted up something to get styles and extended styles from an hWnd:
```py
import os
import win32con
import win32gui
from pywinauto.handleprops import has_exstyle
from pywinauto.win32functions import win32defines


def get_window_styles(hwnd:int, gwl_style_or_exstyle:int, startstr:str):

    style = win32gui.GetWindowLong(hwnd, gwl_style_or_exstyle)
    hex_style = hex(style)
    print(f"style: {style} ({hex_style})")
    print()

    styles = []
    # x = name/key
    # v = value
    for x in dir(win32con):
        v = getattr(win32con, x)
        if callable(v):
            continue
        if not isinstance(v, int):
            continue
        if not x.startswith(startstr):
            continue

        if (style & v) == v:
            styles.append((x, v, hex(v)))

    total = 0
    for (name, value, h) in styles:
        if value == 0:
            continue
        print(f"{name}: {value} ({h})")
        total += value #total += int(h, 16)

    total_hex = hex(total)
    print()
    print(f'total: {total} ({total_hex})')


os.system("cls")
hwnd = 724210 # Replace this with the actual window handle you want to inspect
get_window_styles(hwnd, win32con.GWL_STYLE, "WS_")
print("-"*50)
get_window_styles(hwnd, win32con.GWL_EXSTYLE, "WS_EX_")
exit(0)
```

My output is:
```
style: 369623040 (0x16080000)

WS_CLIPCHILDREN: 33554432 (0x2000000)
WS_CLIPSIBLINGS: 67108864 (0x4000000)
WS_EX_COMPOSITED: 33554432 (0x2000000)
WS_EX_LAYERED: 524288 (0x80000)
WS_SYSMENU: 524288 (0x80000)
WS_VISIBLE: 268435456 (0x10000000)

total: 403701760 (0x18100000)
--------------------------------------------------
style: 524328 (0x80028)

WS_EX_LAYERED: 524288 (0x80000)
WS_EX_TOPMOST: 8 (0x8)
WS_EX_TRANSPARENT: 32 (0x20)

total: 524328 (0x80028)
```
I'm aware that the non-extended style doesn't equal the original given style, in fact it's even larger, which is still unresolved and weird to me.

**Comment by Illegal-Services** [comment]
Actually about the previous post, I think it's pretty trash.
I've writted up something to get styles and extended styles from an hWnd:
```py
import os
import win32con
import win32gui
from pywinauto.handleprops import has_exstyle
from pywinauto.win32functions import win32defines


def get_window_styles(hwnd:int, gwl_style_or_exstyle:int, startstr:str):

    style = win32gui.GetWindowLong(hwnd, gwl_style_or_exstyle)
    hex_style = hex(style)
    print(f"style: {style} ({hex_style})")
    print()

    styles = []
    # x = name/key
    # v = value
    for x in dir(win32con):
        v = getattr(win32con, x)
        if callable(v):
            continue
        if not isinstance(v, int):
            continue
        if not x.startswith(startstr):
            continue
        if x.startswith("WS_EX_"):
            if startstr == "WS_":
                continue

        if (style & v) == v:
            styles.append((x, v, hex(v)))

    total = 0
    for (name, value, h) in styles:
        if value == 0:
            continue
        print(f"{name}: {value} ({h})")
        total += value #total += int(h, 16)

    total_hex = hex(total)
    print()
    print(f'total: {total} ({total_hex})')


os.system("cls")
hwnd = 724210 # Replace this with the actual window handle you want to inspect
get_window_styles(hwnd, win32con.GWL_STYLE, "WS_")
print("-"*50)
get_window_styles(hwnd, win32con.GWL_EXSTYLE, "WS_EX_")
exit(0)
```
My output is:
```
style: 369623040 (0x16080000)

WS_CLIPCHILDREN: 33554432 (0x2000000)
WS_CLIPSIBLINGS: 67108864 (0x4000000)
WS_EX_COMPOSITED: 33554432 (0x2000000)
WS_EX_LAYERED: 524288 (0x80000)
WS_SYSMENU: 524288 (0x80000)
WS_VISIBLE: 268435456 (0x10000000)

total: 403701760 (0x18100000)
--------------------------------------------------
style: 524328 (0x80028)

WS_EX_LAYERED: 524288 (0x80000)
WS_EX_TOPMOST: 8 (0x8)
WS_EX_TRANSPARENT: 32 (0x20)

total: 524328 (0x80028)
```

**Comment by Illegal-Services** [comment]
Just tested also to apply a crosshair on my screen using:
https://github.com/lacymorrow/crossover

it doesn't work too

**Comment by Illegal-Services** [comment]
![image](https://github.com/baffler/Transparent-Twitch-Chat-Overlay/assets/62464560/705f1cc8-ac11-44a3-b9f3-114410415b8e)

i tried reproducing the same styles then the one from the game's window and except MINIMIZE and ICONIC, which for some reasons makes the window dissapear, it's a fail.

**Comment by baffler** [comment]
Possible fix for this in v1.0.2, you can press Ctrl+Alt+F8 to bring the app to the front, while in game. You can change the hotkey as well in the settings.

**Comment by Illegal-Services** [comment]
> Possible fix for this in v1.0.2, you can press Ctrl+Alt+F8 to bring the app to the front, while in game. You can change the hotkey as well in the settings.

Yeah I can tell it fixed for some of my videogames, I uninstalled that one since then but the new version fixed Outlast 1.

Still got the problem on Minecraft but since that's an UWP app thingy it doesn't even surprise me at this point
